### DISCUSSION TITLE
Create users via csv file

### DISCUSSION THREAD

**Discussion by mcvoigt** [drakkan/sftpgo#302] [Ideas]
Hi
Is it possible to create users in bulk using a csv file?
If not, do you find it interesting to have this functionality?

Regards,
Michael Voigt

**Comment by drakkan** [comment]
Hi,

SFTPGo supports loading users and folders from a json dump, you can create a such dump selecting "template" from the web admin. Please try to use it. The generated template can be imported to the same SFTPGo instance or to another. For example you generate and test the dump on a development host and once you are satisfied you import it on your production host. 

I think it is simpler than filling a csv.

I prefer structured data, [here](https://github.com/drakkan/sftpgo/blob/master/httpd/schema/openapi.yaml#L1599) you can find the user schema definition, please try to write a csv that includes different kinds of users (local,S3,GCS, SFTP) and add also some [filters](https://github.com/drakkan/sftpgo/blob/master/httpd/schema/openapi.yaml#L1299) and [sub dir permissions](https://github.com/drakkan/sftpgo/blob/master/httpd/schema/openapi.yaml#L1663), I think you will give up or end up with a really weird csv

**Reply by mcvoigt** [reply]
Hi,
I generated the json file using the template.
How do I import it?

Thank you,
Mic

**Reply by drakkan** [reply]
Select maintenance from the menu on the left and then use the "import" section

![Schermata del 2021-02-07 15-13-54](https://user-images.githubusercontent.com/553263/107149101-1b898280-6957-11eb-9393-d701b5393338.png)

**Reply by mcvoigt** [reply]
Everything worked ok
thank you

Regards,
Mic

**Comment by mcvoigt** [comment]
Hi,
Ok, I’ll try with JSON, I don’t think I’ll have any problems. Warning if
everything goes well.

Regards,
Michael Voigt

Em dom., 7 de fev. de 2021 às 09:53, Nicola Murino <notifications@github.com>
escreveu:
